teuthology/task/internal.py

```diff
@@ -159,6 +159,9 @@ def lock_machines(ctx, config):
             break
         elif not ctx.block:
             assert 0, 'not enough machines are available'
+        else:
+            how_many = how_many - len(newly_locked)
+            assert how_many > 0
```

Comment:
Is this needed?  If it is, can we add a message onto it explaining why the assertion failed?